tests/test_new_staging.py

```diff
@@ -0,0 +1,300 @@
+# Copyright 2024 CS Group
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Test rs-client-libraries staging functions """
+
+import getpass
+import json
+import os
+import os.path as osp
+from pathlib import Path
+from typing import Any
+
+import pytest
+import responses
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_408_REQUEST_TIMEOUT,
+    HTTP_500_INTERNAL_SERVER_ERROR,
+)
+
+from rs_client.rs_client import RsClient
+from rs_common.config import EDownloadStatus
+
+RESOURCES_FOLDER = Path(osp.realpath(osp.dirname(__file__))) / "resources"
+AUXIP = "AUXIP"
+CADIP = "CADIP"
+RESOURCE = "staging"
+APIKEY = None
+OWNER_ID = getpass.getuser()
+OUTPUT_COLLECTION = "my_test_collection"
+
+
+@pytest.fixture(name="rs_server_href")
+def get_rs_server_href():
+    """
+    rs_server_href for local_mode
+    """
+    return None
+
+
+@pytest.fixture(name="staging_client")
+def get_staging_client(rs_server_href):
+    """Create a staging client
+
+    Args:
+        href (str): rs_server href for local-mode
+
+    Returns:
+        StagingClient: StagingClient object to apply the staging
+    """
+    client = RsClient(
+        rs_server_href=rs_server_href,
+        rs_server_api_key=APIKEY,
+        owner_id=OWNER_ID,
+        logger=None,
+    )
+    return client.get_staging_client()
+
+
+@pytest.fixture(name="href_staging")
+def get_href_staging():
+    """
+    Return href for staging
+    """
+    href_staging = os.environ["RSPY_HOST_STAGING"] = "http://127.0.0.1:8004"
```

Comment:
You call `RsClient(rs_server_href=rs_server_href`, then you call `href_staging = os.environ["RSPY_HOST_STAGING"]` to override rs_server_href, then later you mock the responses made when we call `href_staging`. It's simpler to remove this href_staging and just mock rs_server_href.